pkg/controller/jobs/job/job_controller_test.go

```diff
@@ -286,3 +296,172 @@ func TestPodSets(t *testing.T) {
 		})
 	}
 }
+
+var (
+	ignoredJobFields = []cmp.Option{
+		cmpopts.EquateEmpty(),
+		cmpopts.IgnoreFields(batchv1.Job{}, "TypeMeta", "ObjectMeta"),
+	}
+	ignoredWorkloadFields = []cmp.Option{
+		cmpopts.EquateEmpty(),
+		cmpopts.IgnoreFields(kueue.Workload{}, "TypeMeta", "ObjectMeta"),
+		cmpopts.IgnoreFields(metav1.Condition{}, "LastTransitionTime"),
+	}
+)
+
+func TestReconciler(t *testing.T) {
```

Comment:
cc @stuton this is how I envision #922.

For now the tests cases here are very much focused on just equivalency checks, but we can expand.